### ISSUE TITLE
Table column resizing using drag interaction

### ISSUE THREAD

**Issue by andrew-ronaldson** [patternfly/patternfly-design#1119]
## Requesting enhancements to PatternFly table component
- This could be another possible enhancement to the great work being done in tables. Following the addition of horizontal scrolling and column management I propose we offer users the ability to resize table columns by dragging the column edge.
- When working in tables that have several columns the width assigned to each column can become too small and line breaks result in more vertical space used per row. With the addition of horizontal scrolling to PF tables users have more flexibility and this enhancement will help make the data more legible.
- Below is an screenshot of the proposed design change along with a link to an example [here](https://www.ag-grid.com/example.php)
<img width="1147" alt="Screen Shot 2022-01-04 at 3 59 34 PM" src="https://user-images.githubusercontent.com/28658548/148116855-59060c88-c0f8-4451-b0d0-4555608634e8.png">

**Comment by mcarrano** [comment]
Follow up with @tlabaj regarding feasibility

**Comment by mcarrano** [comment]
@tlabaj can you take a look at this when you get a chance and let me know if this is easily supported with our existing React table? And if not, what the scope of work to support it would be? Thanks.

**Comment by tlabaj** [comment]
@mcarrano it look like there is an extension for `reacttabular` called 'reactabular-resizable'.  We could possible pull this in to the table.  I propose creating a react spike issue to investigate the effort and feasibility.

**Comment by tlabaj** [comment]
I have opened a spike issue [#7067](https://github.com/patternfly/patternfly-react/issues/7067) in patternfly-react.

**Comment by mcarrano** [comment]
@andrew-ronaldson we've opened a spike to investigate the technical feasibility of this a bit more before we can commit to it. Please keep us posted if your priority for this changes.

**Comment by mcarrano** [comment]
@andrew-ronaldson We've completed our investigation of the technical feasibility of this. Turns out that it can be added without too much difficulty on top of our legacy (Rectabular) table implementation, but not so easily with our newer composable table approach. You can read more here: https://github.com/patternfly/patternfly-react/issues/7067. Do you have any idea of what your project is using today (for implementation)? What do you feel the priority is for this? I do like the idea of resizable columns but also feels more like a "nice to have." Thoughts?
